theano/tensor/extra_ops.py

```diff
@@ -647,9 +647,36 @@ def repeat(x, repeats, axis=None):
 
     .. versionadded:: 0.6
     """
-    return RepeatOp(axis=axis)(x, repeats)
+    if isinstance(repeats, (int, long, numpy.integer, list)):
+        repeats = numpy.asarray(repeats)
```

Comment:
Well this if-check is for those input that does not have attribute "ndim", like int, list, etc. It is not for theano tensor. I don't need to cast a Tensor to a tensor.vector() in this case,  if repeats is tensor variable, it will automatically have "ndim".